src/template.html

```diff
@@ -108,25 +108,25 @@
             <span class="product-version" title="GraphDB {{mainTitle}}">{{mainTitle}}</span>
         </a>
     </li>
-    <li ng-repeat="item in menu" ng-if="hasRole(item.role)"
+    <li ng-repeat="item in menu track by $index" ng-if="hasRole(item.role)"
         ng-class="{open: $index === selected, clicked: clicked && $index === selected}"
         class="menu-element" ng-click="checkSubMenuPosition($index)">
         <div class="menu-element-root" ng-if="item.children.length" ng-click="select($index, $event, clicked);">
             <span ng-class=item.icon class="menu-item-icon"></span>
             <span class="menu-item">{{item.label}}</span>
         </div>
-        <a class="menu-element-root" ng-href="{{item.hrefFun ? item.hrefFun() : item.href}}" ng-if="!item.children.length"
+        <a class="menu-element-root" ng-href="{{::item.hrefFun ? item.hrefFun(productInfo) : item.href}}" ng-if="!item.children.length"
```

Comment:
Added one time binding here an to the submenus below to prevent perpetual calculation of the hrefs on every digest cycle.